src/linear_algebra/tensor_product.lean

```diff
@@ -713,6 +713,18 @@ let e₁ := tensor_product.assoc R M N (P ⊗[R] Q),
     e₃ := (tensor_product.assoc R M P (N ⊗[R] Q)).symm in
 e₁ ≪≫ₗ (e₂ ≪≫ₗ e₃)
 
+/-- This special case is useful for describing the interplay between `dual_tensor_hom_equiv` and
+composition of linear maps.
+
+E.g., composition of linear maps gives a map `(M → N) ⊗ (N → P) → (M → P)`, and applying
+`dual_tensor_hom_equiv.symm` to the three hom-modules gives a map
+`(M.dual ⊗ N) ⊗ (N.dual ⊗ P) → (M.dual ⊗ P)`, which agrees with the application of `contract_right`
+on `N ⊗ N.dual` after the suitable rebracketting.
+-/
+def tensor_tensor_tensor_assoc : (M ⊗[R] N) ⊗[R] (P ⊗[R] Q) ≃ₗ[R] M ⊗[R] (N ⊗[R] P) ⊗[R] Q :=
+(tensor_product.assoc R (M ⊗[R] N) P Q).symm ≪≫ₗ
+congr (tensor_product.assoc R M N P) (1 : Q ≃ₗ[R] Q)
```

Comment:
I think it would be better to move this below `tensor_tensor_tensor_comm_symm_tmul` to keep the lemmas adjacent to their definitions